### PR TITLE
14680 More debug info

### PR DESCRIPTION
**Pull request by Sergio-Mira**
https://swrvedev.jira.com/browse/SWRVE-14680
Adds better data migration to Application data.
Remove some internal events from trying to trigger IAM campaigns.

### CODE REVIEW

**Comment by Sergio-Mira** [comment]
@adam-govan @seaders @dominicmarmionswrve Could you have a quick look?

File: SwrveSDK/SDK/Track/Swrve.m [hunk 1]
```diff
@@ -1915,6 +1939,16 @@ - (void) queueDeviceProperties
     [self userUpdate:self.deviceInfo];
 }
 
+- (BOOL) canSendExtraLogs {
+    return self->extraLogs;
+}
+
+- (void) extraLogEvent:(NSString*)name payload:(NSDictionary*)payload {
+    if (self->extraLogs) {
+        [self eventInternal:[@"Swrve.extra_log." stringByAppendingString:name] payload:payload triggerCallback:NO];
```
Comment (johngilbert) [inline SwrveSDK/SDK/Track/Swrve.m]: Similar to the comment on https://github.com/Swrve/swrve/pull/25526, I think we need a more specific name than 'extra_log' to future-proof ourselves.. possibly 'Swrve.debug.ios_campaign_state_log'?

**Comment by johngilbert** [comment]
It looks like we're sending 11 new regular events with a naming convention of 'Swrve.extra_log.XYZ'

campaign_state_from_disk
campaign_state_serialization_error
campaign_state
campaign_state_write_error
application_support_error
install_time_primary_file_read_error
install_time_primary_file_write_error
install_time_secondary_file_read_error
install_time_write_error
file_migration_old_file_delete_failed
file_migration_failed

Each of these will generate reports for histogram payloads, timeseries, be exported in the UserDBs, appear in the event-reports page in the dashboard, consume space in the EP's bdbs etc. 

I presume we just want these to get into the event log so that we can pull the raw events via slipstream or similar for offline analysis, if so we need to restrict processing of these '.extra_log' events through the system?

How does this logging infrastructure / naming conventions sit with our other supported platforms?

File: SwrveSDK/SDK/Track/Swrve.m [hunk 2]
```diff
@@ -1915,6 +1939,16 @@ - (void) queueDeviceProperties
     [self userUpdate:self.deviceInfo];
 }
 
+- (BOOL) canSendExtraLogs {
+    return self->extraLogs;
+}
+
+- (void) extraLogEvent:(NSString*)name payload:(NSDictionary*)payload {
```
Comment (johngilbert) [inline SwrveSDK/SDK/Track/Swrve.m:1941]: Should this have a (reasonable) level of rate limiting if someone is doing something crazy with their integration? I'm not sure what the worst-case-scenario that's possible is right now with an integration / set of active campaigns for us here. We don't want an unbounded amount of debug logging to be sent, for all users, to our sinks if we flip a feature flag.

**Pull request closed without merging**